docs/resources/app.md

```diff
@@ -0,0 +1,105 @@
+---
+subcategory: "Apps"
+---
+# databricks_app Resource
+
+-> This feature is in [Public Preview](https://docs.databricks.com/release-notes/release-types.html).
+
+Apps run directly on a customer’s Databricks instance, integrate with their data, use and extend Databricks services, and enable users to interact through single sign-on.
```

Comment:
We should probably mention that this only creates the application and doesn't do anything with the app deployment.